obspy/core/event.py

```diff
@@ -756,10 +756,17 @@ def get_quakeml_uri(self, authority_id="local"):
 
         regex = r"^(smi|quakeml):[\w\d][\w\d\-\.\*\(\)_~']{2,}/[\w\d\-\." + \
                 r"\*\(\)_~'][\w\d\-\.\*\(\)\+\?_~'=,;#/&amp;]*$"
+        # if current string is a valid QuakeML resource ID just use it
         result = re.match(regex, str(id))
         if result is not None:
             return id
-        id = 'smi:%s/%s' % (authority_id, str(id))
+        # if not, we replace invalid characters and prepend a valid prefix
+        id_string, replace_count = \
+            re.subn(r"[^\w\d\-\.\*\(\)\+\?_~'=,;#/&amp;]", '_', str(id))
```

Comment:
So I realize this was copied out of the spec, but I don't think `&amp;` was intended here or above.